builtins: implement ST_Simplify and ST_SimplifyPreserveTopology

otan opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/49036
Resolves https://github.com/cockroachdb/cockroach/issues/49038
Refs https://github.com/cockroachdb/cockroach/issues/52013

Release justification: low risk change to new functionality

Release note (sql change): Implement ST_Simplify and
ST_SimplifyPreserveTopology.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53796)
<!-- Reviewable:end -->

otan committed:
builtins: implement ST_Simplify and ST_SimplifyPreserveTopology

Release justification: low risk change to new functionality

Release note (sql change): Implement ST_Simplify and
ST_SimplifyPreserveTopology.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 8 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53796#-:-MGFUnkd0TW8xflijzjM:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola

otan commented:
bors r=sumeerbhola
